core/CMakeLists.txt

```diff
@@ -64,18 +64,22 @@ add_library(Ginkgo::ginkgo ALIAS ginkgo)
 target_link_libraries(ginkgo
     PUBLIC ginkgo_device ginkgo_omp ginkgo_cuda ginkgo_reference ginkgo_hip ginkgo_dpcpp)
 # The PAPI dependency needs to be exposed to the user.
+set(GKO_RPATH_ADDITIONS "")
 if (GINKGO_HAVE_PAPI_SDE)
     target_link_libraries(ginkgo PUBLIC PAPI::PAPI)
+    list(GET PAPI_LIBRARIES 0 PAPI_FIRST_LIB)
+    get_filename_component(GKO_PAPI_LIBDIR "${PAPI_FIRST_LIB}" DIRECTORY)
+    list(APPEND GKO_RPATH_ADDITIONS "${GKO_PAPI_LIBDIR}")
 endif()
 
-# Since we have a public dependency on HIP, this dependency appears here as well
-set(RPATH_ADD_HIP_PATH "")
+# Since we have a public dependency on HIP, this dependency appears
+# here as well
 if (GINKGO_BUILD_HIP AND GINKGO_HIP_PLATFORM MATCHES "hcc")
-    set(RPATH_ADD_HIP_PATH "${HIP_PATH}/lib")
+    list(APPEND GKO_RPATH_ADDITIONS "${HIP_PATH}/lib")
```

Comment:
Do we always add this to the RPATH, or just in case GINKGO_INSTALL_RPATH_DEPENDENCIES is ON?

Reply:
Yes, this is passed to the `ginkgo_install_library` which gets then passed to the `ginkgo_add_install_rpath`, and nothing gets done when we are not either in `BUILD_SHARED_LIBS` or `GINKGO_INSTALL_RPATH`.
https://github.com/ginkgo-project/ginkgo/blob/fix_hip/cmake/install_helpers.cmake#L27-L30

Reply:
I think it already did. it passed by ARGN.

Reply:
I will add more description on that topic.